src/TLuaInterpreter.cpp

```diff
@@ -10483,81 +10531,81 @@ int TLuaInterpreter::echoPopup(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#echoLink
 int TLuaInterpreter::echoLink(lua_State* L)
 {
-    string a1;
-    string a2;
-    string a3;
-    string a4;
-    bool a5 = false;
+    QString a1;
+    QString a2;
+    QString a3;
+    QString a4;
+    bool useCurrentFormatElseDefault = false;
     bool gotBool = false;
 
     int s = 1;
     int n = lua_gettop(L);
 
     if (!lua_isstring(L, s)) {
-        lua_pushstring(L, "echoLink: wrong argument type");
+        lua_pushfstring(L, "echoLink: bad argument #%d type (string expected, got %s!)", s, luaL_typename(L, s));
         lua_error(L);
         return 1;
     } else {
-        a1 = lua_tostring(L, s);
+        a1 = QString::fromUtf8(lua_tostring(L, s));
         s++;
     }
     if (n > 1) {
         if (!lua_isstring(L, s)) {
-            lua_pushstring(L, "echoLink: wrong argument type");
+            lua_pushfstring(L, "echoLink: bad argument #%d type (string expected, got %s!)", s, luaL_typename(L, s));
             lua_error(L);
             return 1;
         } else {
-            a2 = lua_tostring(L, s);
+            a2 = QString::fromUtf8(lua_tostring(L, s));
             s++;
         }
     }
     if (n > 2) {
         if (!lua_isstring(L, s)) {
-            lua_pushstring(L, "echoLink: wrong argument type");
+            lua_pushfstring(L, "echoLink: bad argument #%d type (string expected, got %s!)", s, luaL_typename(L, s));
             lua_error(L);
             return 1;
         } else {
-            a3 = lua_tostring(L, s);
+            a3 = QString::fromUtf8(lua_tostring(L, s));
             s++;
         }
     }
     if (n > 3) {
         if (lua_isstring(L, s)) {
-            a4 = lua_tostring(L, s);
+            a4 = QString::fromUtf8(lua_tostring(L, s));
             s++;
         } else if (lua_isboolean(L, s)) {
             gotBool = true;
-            a5 = lua_toboolean(L, s);
+            useCurrentFormatElseDefault = lua_toboolean(L, s);
             s++;
         }
     }
```

Comment:
:thinking: If it isn't a string or a boolean do we need to report something here?

Reply:
applied.